frontend/src/components/pages/roles/UserAclsCard.test.tsx

```diff
@@ -0,0 +1,137 @@
+      {
+        id: 1,
+        resourceType: 'topic',
+        mode: 'custom',
+        selectorType: 'literal',
+        selectorValue: 'test-topic',
+        operations: {
+          READ: 'allow',
+          WRITE: 'allow',
+        },
+      },
+    ],
+  },
+  {
+    sharedConfig: {
+      principal: 'User:test-user',
+      host: '192.168.1.1',
+    },
+    rules: [
+      {
+        id: 2,
+        resourceType: 'cluster',
+        mode: 'custom',
+        selectorType: 'any',
+        selectorValue: '',
+        operations: {
+          DESCRIBE: 'allow',
+        },
+      },
+    ],
+  },
+];
+
+describe('UserAclsCard', () => {
+  it('should render empty state when no ACLs provided', () => {
+    render(
+      <BrowserRouter>
+        <UserAclsCard acls={[]} />
+      </BrowserRouter>,
+    );
+
+    expect(screen.getByText('ACLs (0)')).toBeInTheDocument();
+    expect(screen.getByText('No ACLs assigned to this user.')).toBeInTheDocument();
+    expect(screen.getByRole('button', { name: 'Create ACL' })).toBeInTheDocument();
+  });
+
+  it('should render empty state when acls is undefined', () => {
+    render(
+      <BrowserRouter>
+        <UserAclsCard acls={undefined} />
+      </BrowserRouter>,
+    );
+
+    expect(screen.getByText('ACLs (0)')).toBeInTheDocument();
+    expect(screen.getByText('No ACLs assigned to this user.')).toBeInTheDocument();
+    expect(screen.getByRole('button', { name: 'Create ACL' })).toBeInTheDocument();
+  });
+
+  it('should render ACL table with correct count', () => {
+    render(
+      <BrowserRouter>
+        <UserAclsCard acls={mockAcls} />
+      </BrowserRouter>,
+    );
+
+    expect(screen.getByText('ACLs (2)')).toBeInTheDocument();
+  });
+
+  it('should render ACL rows with principal and host', () => {
+    render(
+      <BrowserRouter>
+        <UserAclsCard acls={mockAcls} />
+      </BrowserRouter>,
+    );
+
+    expect(screen.getByTestId('acl-principal-User:test-user-*')).toHaveTextContent('User:test-user');
+    expect(screen.getByTestId('acl-principal-User:test-user-192.168.1.1')).toHaveTextContent('User:test-user');
+    expect(screen.getByTestId('acl-host-*')).toHaveTextContent('*');
+    expect(screen.getByTestId('acl-host-192.168.1.1')).toHaveTextContent('192.168.1.1');
+  });
+
+  it('should render action buttons for each ACL', () => {
+    render(
+      <BrowserRouter>
+        <UserAclsCard acls={mockAcls} />
+      </BrowserRouter>,
+    );
+
+    expect(screen.getByTestId('view-acl-User:test-user-*')).toBeInTheDocument();
+    expect(screen.getByTestId('view-acl-User:test-user-192.168.1.1')).toBeInTheDocument();
+  });
+
+  it('should render table headers', () => {
+    render(
+      <BrowserRouter>
+        <UserAclsCard acls={mockAcls} />
+      </BrowserRouter>,
+    );
+
+    expect(screen.getByText('Name')).toBeInTheDocument();
+    expect(screen.getByText('Hosts')).toBeInTheDocument();
+    expect(screen.getByText('Actions')).toBeInTheDocument();
+  });
+});
```

Comment:
Similar to the previous tests, let's just do 1 test for ACLs view with rows and 1 test for empty state, I think the other tests in this file wouldn't increase my confidence in releasing